docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -269,10 +269,41 @@ CertificateCredential credential = identity.getCredential(CertificateCredential.
 X509Certificate certificate = credential.getCertificate();
 ----
 
-==== Authorization
+[[map-certificate-attributes-to-roles]]
+==== Mapping certificate attributes to roles
+
+The information from the client certificate can be used to add roles to Quarkus `SecurityIdentity`.
+
+You can add new roles to `SecurityIdentity` after checking a client certificate's common name(CN) attribute.
+The easiest way to do is to use a certificate attribute to role mapping feature.
+
+For example, you can update the properties shown in the section which inroduces <<mutual-tls>> as follows:
+
+[source,properties]
+----
+quarkus.http.ssl.certificate.key-store-file=server-keystore.jks
+quarkus.http.ssl.certificate.key-store-password=the_key_store_secret
+quarkus.http.ssl.certificate.trust-store-file=server-truststore.jks
+quarkus.http.ssl.certificate.trust-store-password=the_trust_store_secret
+quarkus.http.ssl.client-auth=required
+quarkus.http.insecure-requests=disabled
+
+quarkus.http.auth.certificate-role-properties=cert-role-mappings.properties <1>
+
+quarkus.http.auth.permission.certauthenticated.paths=/*   <2>
+quarkus.http.auth.permission.certauthenticated.policy=role-policy-cert <2>
+quarkus.http.auth.policy.role-policy-cert.roles-allowed=user,admin     <2>
+----
+<1> The `cert-role-mappings.properties` classpath resource contains a map of certificate's CN values to roles in the form `CN=role` or `CN=role1,role2`, etc. Let's assume it contains three entries: `alice=user,admin`, `bob=user` and `jdoe=tester`.
```

Comment:
You can also map whole principal as well. I don't ask you to add it is as I'm not all that sure it's practical. Please consider whether you want to document it or not. Your call.

Reply:
@michalvavrik This doc update is only about mapping the CN attribute to roles (a bit later SAN or other attributes as well), to support QUARKUS-3466. 

But, possibly for the follow up doc update,

>You can also map whole principal as well

Can you clarify this point ?

Reply:
https://github.com/quarkusio/quarkus/blob/70c26711ce14500595276d120ffb94f346aebdc7/extensions/security/runtime/src/main/java/io/quarkus/security/runtime/X509IdentityProvider.java#L49

means you can do this

https://github.com/quarkus-qe/quarkus-test-suite/blob/2c618a33303af619ba736b3e07ea6499d9b70b7a/security/https/src/main/resources/role-mappings.txt

Reply:
It doesn't mean you should, I just tried to remind you in case you wanted to document this...

Reply:
Thanks @michalvavrik Yes, I recall now, not sure why I typed that code, but it can be handy I guess for some complex CNames when the CName calculation may not produce the correct value, but I don't really know when the CName calculation may fail, so we should probably doc this extra option when we have a good reason to recommend having to type the complete expression `CN=...`.

Reply:
Thanks @michalvavrik for reminding we can type it